staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/lastappliedupdater_test.go

```diff
@@ -89,3 +93,134 @@ spec:
 		t.Errorf("expected last applied annotation to be updated, but got: %q", lastApplied)
 	}
 }
+
+func TestLargeLastApplied(t *testing.T) {
+	tests := []struct {
+		name      string
+		oldObject *corev1.ConfigMap
+		newObject *corev1.ConfigMap
+	}{
+		{
+			name: "old object + new object last-applied annotation is too big",
+			oldObject: &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "v1",
+					Kind:       "ConfigMap",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "large-update-test-cm",
+					Namespace: "default",
+					Annotations: map[string]string{
+						corev1.LastAppliedConfigAnnotation: "nonempty",
```

Comment:
I'm surprised we're updating an invalid CSA annotation to a valid one... shouldn't we just be dropping it in this case?

Reply:
Currently we check if the annotation simply exists, but we could try to decode the annotation and otherwise drop it if it's invalid

Reply:
how would someone wanting to transition from CSA to SSA and clear the annotation for space reasons do it with apply? wouldn't they have to set it to "" in their SSA request? we should at least check that the old annotation is non-empty.

Reply:
Yeah that sounds like a reasonable check. Currently, users would clear the annotation to opt out of this behavior, but setting it to "" sounds like they're also expressing that they want to opt-out of the behavior. I'll add a non-empty check.

Reply:
> Currently, users would clear the annotation to opt out of this behavior

there's no way to clear an annotation you don't own with a server-side apply other than setting it to "", right?

Reply:
Oh right, when you don't have exclusive ownership. And in this case, CSA and SSA would always share ownership. Related: https://github.com/kubernetes/kubernetes/issues/99003